bakery/checkers/namespace.go

```diff
@@ -0,0 +1,94 @@
+package checkers
+
+// Namespace holds maps from schema URIs to the
+// prefixes that are used to encode them in first party
+// caveats. Several different URIs may map to the same
+// prefix - this is usual when several different backwardly
+// compatible schema versions are registered.
+type Namespace struct {
+	uriToPrefix map[string]string
+}
+
+// NewNamespace returns a new namespace with the
+// given initial contents.
+func NewNamespace(uriToPrefix map[string]string) *Namespace {
+	ns := &Namespace{
+		uriToPrefix: make(map[string]string),
+	}
+	for uri, prefix := range uriToPrefix {
+		ns.uriToPrefix[uri] = prefix
+	}
+	return ns
+}
+
+// EnsureResolved tries to resolve the given schema URI to a prefix and
+// returns the prefix and whether the resolution was successful. If the
+// URI hasn't been registered but a compatible version has, the
+// given URI is registered with the same prefix.
+func (ns *Namespace) EnsureResolved(uri string) (string, bool) {
+	// TODO compatibility
+	return ns.Resolve(uri)
+}
+
+// Resolve resolves the given schema URI to its registered prefix and
+// returns the prefix and whether the resolution was successful.
+//
+// If ns is nil, it is treated as if it were empty.
+//
+// Resolve does not mutate ns and may be called concurrently
+// with other non-mutating Namespace methods.
+func (ns *Namespace) Resolve(uri string) (string, bool) {
+	if ns == nil {
+		return "", false
+	}
+	prefix, ok := ns.uriToPrefix[uri]
```

Comment:
return ns.uriToPrefix[uri]? (Does that work, is go able to infer you want to get from a map with bool from the return type of this function?